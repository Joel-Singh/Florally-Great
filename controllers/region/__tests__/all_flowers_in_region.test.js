const path = require("path");

const Region = require(path.join(appRoot, "models", "region.js"));
const Flower = require(path.join(appRoot, "models", "flower.js"));

const all_flowers_in_region = require("../all_flowers_in_region.js");

test("Passes in just region", async () => {
  const regionName = "regionName";
  const fakeReq = {
    params: {
      name: regionName,
    },
  };

  const fakeRes = {
    render: jest.fn(),
  };

  await new Region({
    name: regionName,
    description: "description",
    _id: "6483d106cdcd7f4f8d6ed46a",
  }).save();

  await all_flowers_in_region(fakeReq, fakeRes);

  expect(fakeRes.render.mock.calls).toMatchInlineSnapshot(`
    [
      [
        "regions/all_flowers_in_region",
        {
          "flower_list": [],
          "region": {
            "__v": 0,
            "_id": "6483d106cdcd7f4f8d6ed46a",
            "description": "description",
            "name": "regionName",
          },
        },
      ],
    ]
  `);
});

test("Passes in region and flowers", async () => {
  const regionName = "regionName";
  const regionId = "6483d106cdcd7f4f8d6ed46a";

  const fakeReq = {
    params: {
      name: regionName,
    },
  };

  const fakeRes = {
    render: jest.fn(),
  };

  await new Region({
    name: regionName,
    description: "description",
    _id: "6483d106cdcd7f4f8d6ed46a",
  }).save();

  await new Flower({
    name: "Flower in region",
    description: "description",
    price: 32,
    numberInStock: 92,
    region: regionId,
    _id: "6487734ae98dc419fc0f170d"
  }).save();

  await all_flowers_in_region(fakeReq, fakeRes);

  expect(fakeRes.render.mock.calls).toMatchSnapshot();
});
