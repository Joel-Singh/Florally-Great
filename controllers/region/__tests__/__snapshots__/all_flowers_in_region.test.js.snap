// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passes in region and flowers 1`] = `
[
  [
    "regions/all_flowers_in_region",
    {
      "flower_list": [
        {
          "__v": 0,
          "_id": "6487734ae98dc419fc0f170d",
          "description": "description",
          "name": "Flower in region",
          "numberInStock": 92,
          "price": 32,
          "region": "6483d106cdcd7f4f8d6ed46a",
        },
      ],
      "region": {
        "__v": 0,
        "_id": "6483d106cdcd7f4f8d6ed46a",
        "description": "description",
        "name": "regionName",
      },
    },
  ],
]
`;
